src/help.cpp

```diff
@@ -157,136 +115,216 @@ std::string help::get_note_colors()
-
-                if( !i18n_help_texts.empty() ) {
-                    ui.on_screen_resize( nullptr );
-
-                    const auto get_w_help_border = [&]() {
-                        init_windows( ui );
-                        return w_help_border;
-                    };
-
-                    scrollable_text( get_w_help_border, _( "Help" ),
-                                     std::accumulate( i18n_help_texts.begin() + 1, i18n_help_texts.end(),
-                                                      i18n_help_texts.front(),
-                    []( std::string lhs, const std::string & rhs ) {
-                        return std::move( lhs ) + "\n\n" + rhs;
-                    } ) );
-
-                    ui.on_screen_resize( init_windows );
+void help_window::draw_category_option( const int &option, const help_category &category )
+{
+    std::string cat_name;
+    auto hotkey_it = hotkeys.find( option );
+    if( hotkey_it != hotkeys.end() ) {
+        cat_name = hotkey_it->second.short_description();
+        cat_name += ": ";
+    }
+    cat_name += category.name.translated();
+    if( data.read_categories.find( option ) != data.read_categories.end() ) {
+        ImGui::PushStyleColor( ImGuiCol_Text, c_light_gray );
+        ImGui::Selectable( remove_color_tags( cat_name ).c_str() );
+        ImGui::PopStyleColor();
+    } else {
+        ImGui::Selectable( remove_color_tags( cat_name ).c_str() );
+    }
+    if( ImGui::IsItemHovered() ) {
+        selected_option = option;
+    }
+}
+
+std::string help_window::seperator( int length, char c )
+{
+    std::string ret = "<color_light_blue>";
+    ret += std::string( length, c );
+    ret += "</color>";
+    return ret;
+}
+
+void help_window::format_title( const std::string translated_category_name )
+{
+    cataimgui::PushMonoFont();
+    if( get_option<bool>( "SCREEN_READER_MODE" ) ) {
+        cataimgui::draw_colored_text( translated_category_name );
+        ImGui::NewLine();
+        return;
+    }
+    std::string row3 = "<color_light_blue>║</color> ";
+    row3 += translated_category_name;
+    row3 += " <color_light_blue>║</color>";
+    std::string div = seperator( remove_color_tags( row3 ).length() - 4, '=' );
+    cataimgui::draw_colored_text( div );
+    cataimgui::draw_colored_text( row3 );
+    cataimgui::draw_colored_text( div );
+    cataimgui::draw_colored_text( seperator( TERMX, '_' ) );
+    ImGui::NewLine();
+    ImGui::PopFont();
+}
+
+void help_window::draw_category()
+{
+    const help_category &cat = data.help_categories[loaded_option];
+    format_title( cat.name.translated() );
+    if( ImGui::BeginTable( "HELP_PARAGRAPHS", 1,
+                           ImGuiTableFlags_ScrollY ) ) {
+        ImGui::TableNextRow();
+        ImGui::TableNextColumn();
+        for( const translation &paragraph : cat.paragraphs ) {
+            std::string translated_paragraph = paragraph.translated();
+            parse_tags_help_window( translated_paragraph );
```

Comment:
Since translating the paragraph and then parsing the help tags allocates, could you stash the result in a member variable and reuse it instead of doing it again every frame?